src/parser/core/modules/CooldownDowntime.js

```diff
@@ -0,0 +1,183 @@
+			if (dt.current) {
+				dt.current.stoptime = endTime
+				dt.history.push(dt.current)
+				dt.current = null
+			}
+
+			//calculate the downtime based on the start and stop values and sum the array
+			//Adjust for the classes defined alloted time to allow a CD to be held
+			//this supports classes like RDMs who routinely hold CDs due to procs
+			const totalSumOfDownTime = dt.history.map(downTime => {
+				return this._getDownTimeAdjustedForInvuln(downTime.starttime, downTime.stoptime)
+				//return Math.max(downTime.stoptime - downTime.starttime - this._downtimeOkTime, 0)
+			}).reduce(
+				(accumulator, currentValue) => accumulator + currentValue
+			)
+			//write the results as a new Requirement to show up later
+			OGCDRequirements.push(
+				new Requirement({
+					name: <ActionLink {...getAction(id)} />,
+					percent: this._percentFunction(id, totalSumOfDownTime, encounterLength),
+				})
+			)
+
+		})
+
+		//new Rule and adds the array of Requirements that just got generated
+		this.checklist.add(new Rule({
+			name: 'Use your OGCDs',
+			description: `Always make sure to use your OGCDs when they are up but don't clip them.  ${this._downtimeOkTime === 0 ? '' : `To account for random factors you are given a buffer of ${this._downtimeOkTime/1000} seconds per instance to use your CD`}`,
+			requirements: OGCDRequirements,
+			target: 95,
+		}))
+	}
+
+	/**
+	 * Gets the Downtime of a CD adjusted for boss invuln windows as well as globally allowed
+	 * buffer set by the extender class module
+	 * @param {Start Time of the CD Downtime} startTime
+	 * @param {Stop Time of the CD Downtime} stopTime
+	 */
+	_getDownTimeAdjustedForInvuln(startTime, stopTime) {
+		const invuln = this.downtime.getDowntime(startTime, stopTime)
+		return Math.max(stopTime - startTime - invuln - this._downtimeOkTime, 0)
+	}
+
+	//cool function that Furst invented that just sets ok usage as 98% and falls very quickly
+	_percentFunction(actionId, downtime, fightlength) {
+		const cooldown = getAction(actionId).cooldown
+		const possibleNumberOfUses = Math.floor(fightlength/(cooldown*1000))
+		return ((possibleNumberOfUses - Math.floor(downtime/(cooldown*1000)))/possibleNumberOfUses)*100
+	}
+
+	_cooldownCheck(event) {
+		const actionId = event.ability.guid
+
+		//if we get one of the OGCDs stop downtime the old downtime and start a new one
+		if (this._trackedcds.includes(actionId)) {
+			this._stopDowntime(actionId)
+			this._startDowntime(actionId)
+		}
+	}
+
+	_getDowntime(actionId) {
+		return this._downTime[actionId] || {
+			current: null,
+			history: [],
+		}
+	}
+
+	_startDowntime(actionId) {
+		const action = getAction(actionId)
+
+		//generating a new downtime or pshing the old in history
+		const dt = this._getDowntime(actionId)
+		if (dt.current) {
+			dt.history.push(dt.current)
+		}
+
+		//start of downtime based on CD of the OGCD and save it
+		dt.current = {
+			starttime: this.parser.currentTimestamp + action.cooldown * 1000,
```

Comment:
`start`/`end`

Also, this is essentially a lookahead. Consider cases like WAR's Infuriate, which gets CDR from gauge abilities. This doesn't account for that.

Reply:
This is in core, I can't possibly rely on gauge data for a class module which exists above this.  I don't know how many corner cases like that there are, but that won't be supported until another pass that allows overriding at each stage.